core/trace/dt_span_enricher.go

```diff
@@ -36,6 +38,7 @@ func createSpanMetadata(
 
 	metadata := newDtSpanMetadata(spanProcessingIntervalMs)
 	metadata.tenantParentSpanId = tenantParentSpanIdFromContext(parentCtx)
+	metadata.propagatedResourceAttributes = getPropagatedResourceAttributes(parentCtx)
```

Comment:
Consider also storing the attributes on the parent in the case of  a direct child of a (local) root. Should improve performance for subsequent children and may prevent some inconsistencies. Although it could then require special-casing local roots in serialization to not lose the performance again.

Reply:
i think this would also need some additional synchronization/locking since multiple children could potentially try to update the (local) root in parallel. so i'd rather prefer to leave it as is for now.

Reply:
Ah, this happens on the application thread and not the worker goroutine? Then I agree that the current way, requiring no locking, is definitely better.